### DISCUSSION TITLE
Techmap principle

### DISCUSSION THREAD

**Discussion by quantrpeter** [YosysHQ/yosys#3687] [Q&A]
Hi
   Is there any paper/web/anything talk about how yosys do technology mapping? Especially map to FPGA primitives?
thanks
Peter

**Comment by tomverbeure** [comment]
My blog post discusses the basics, including how it maps multiplication onto an FPGA DSP..

https://tomverbeure.github.io/2022/11/18/Primitive-Transformations-with-Yosys-Techmap.html

**Comment by quantrpeter** [comment]
thanks @tomverbeure . I alway search your article out from google, your blog is useful and famous, thanks for the sharing
